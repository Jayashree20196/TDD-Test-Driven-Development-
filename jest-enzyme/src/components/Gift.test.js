import React from "react";
import { shallow } from 'enzyme';
import Gift from './Gift';

describe('Gift Component', () => {
    const gift = shallow(<Gift/>)

    it('renders correctly', () => {
        expect(gift).toMatchSnapshot();
    });

    it('initializes for present and person from state', () => {
        expect(gift.state()).toEqual({person: '', present: ''});
    });

    describe('for input the person name', () => {
        const person = 'uncle';

       beforeEach(() => {
            gift.find('.input-person').simulate('change', { target : {
                value : person
                }
            })
        })

        it('input the person name', () => {
            expect(gift.state().person).toEqual(person)
        })
    })

    describe('for input the present name', () => {
        const present = 'present';

        beforeEach(() => {
            gift.find('.input-present').simulate('change', {target : {
                value:present
            }})
        })

        it('input the present name', () => {
            expect(gift.state().present).toEqual(present)
        }) 
    })

   
})