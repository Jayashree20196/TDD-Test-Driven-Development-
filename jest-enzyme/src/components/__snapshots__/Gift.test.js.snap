// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Component renders correctly 1`] = `ShallowWrapper {}`;
